packages/dotcom-ui-data-embed/README.md

```diff
@@ -0,0 +1,88 @@
+# @financial-times/dotcom-ui-data-embed
+
+This package provides methods for putting data into your server-side rendered pages and safely retrieving it again in the browser.
+
+## Getting started
+
+This package is compatible with Node 12+ and is distributed on npm.
+
+```sh
+npm install --save @financial-times/dotcom-ui-data-embed
+```
+
+After installing the package you can use it to embed data into your pages on the server-side. This data can then be retrieved and used in your client-side code.
+
+### Server-side integration
+
+If you are using React to render your app you can use the `DataEmbed` component to integrate the data embed data with your pages:
+
+```jsx
+import { DataEmbed } from '@financial-times/dotcom-ui-data-embed'
+const DATA_EMBED_ID = 'data-embed'
+const data = { property: 'value', secondProperty: 'second-value' }
+
+export default (props) => (
+  <html>
+    <head>
+      <meta charSet="utf-8" />
+      <title>My Amazing Website</title>
+      <DataEmbed id={DATA_EMBED_ID} data={data} />
+    </head>
+    <body>
+      ...
+    </body>
+  </html>
+)
+```
+
+Otherwise you can insert a JSON formatted string into a `<script>` element with an ID of `data-embed`.
+
+```html
+<!DOCTYPE html>
+    <html>
+    <head>
+        <meta charset="utf-8">
+        <title>My Amazing Website</title>
+        <script type="application/json" id="data-embed">
+        { "property":"value","secondProperty":"second-value"}
+        </script>
+    </head>
+    <body>
+        ...
+    </body>
+</html>
+```
+
+### Client-side integration
+
+Once you have you data embedded in your page you can use the [data embed client](#client-side-api) in your client-side code. The client provides methods for safely retrieving the status of individual context properties.
+
+```js
+import * as dataEmbed from '@financial-times/dotcom-ui-data-embed'
+
+const dataEmbedClient = dataEmbed.init({ id: 'data-embed' })
+
+if (dataEmbedClient.get('property')) {
```

Comment:
I think I'm mostly worried about the performance cost of shipping a deep-freeze library and I'm not too sure of the consequences of us not doing this. I suppose we want to encourage people to make copies of the data rather than relying on the ability to mutate it deeply...

Should we investigate the bundle cost of doing a deep-freeze?

Reply:
In terms of freezing flags and appContext, that should be unchanged as they don't reuse the DataEmbedStore.

Reply:
Let's keep it just as a `.freeze` as `flags` and `appContext` currently are and worry about it if the nested properties being unfrozen ever becomes an issue.

I asked more for my curiosity.